candidates/templates/candidates/posts.html

```diff
@@ -19,7 +19,7 @@
         <h3>{{ role_data.role }}</h3>
         {% for election_data in role_data.elections %}
           {% with election=election_data.election %}
-            <h4>{{ election.name }}</h4>
+            <a href="{% url 'constituencies' election=election.slug %}"><h4>{{ election.name }}</h4></a>
             <ul>
```

Comment:
I'd put the <a> inside the <h4> because otherwise the link is still black text and not at all obvious.